source/process_common.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/process.h>
+#include <aws/common/string.h>
+
+#include <stdio.h>
+#include <sys/types.h>
+
+#define MAX_BUFFER_SIZE (2048)
+
+int aws_run_command_result_init(struct aws_allocator *allocator, struct aws_run_command_result *result) {
+    if (!allocator || !result) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+    AWS_ZERO_STRUCT(*result);
+    return AWS_OP_SUCCESS;
+}
+
+void aws_run_command_result_cleanup(struct aws_run_command_result *result) {
+    if (!result) {
+        return;
+    }
+    aws_string_destroy_secure(result->std_out);
+    aws_string_destroy_secure(result->std_err);
+}
+
+int aws_run_command(
+    struct aws_allocator *allocator,
+    struct aws_run_command_options *options,
+    struct aws_run_command_result *result) {
+
+    AWS_FATAL_ASSERT(allocator);
+    AWS_FATAL_ASSERT(options);
+    AWS_FATAL_ASSERT(result);
+
+    FILE *output_stream;
+    char output_buffer[MAX_BUFFER_SIZE];
+    struct aws_byte_buf result_buffer;
+    bool success = false;
```

Comment:
Minor nit but I always like to just have the return value be a variable that starts AWS_OP_ERR and changes to AWS_OP_SUCCESS, which turns the "if (success) return AWS_OP_SUCCESS; return AWS_OP_ERR" statements into just "return result;"